Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -154,16 +154,40 @@ GridArea GridMasonryLayout::gridAreaForDefiniteGridAxisItem(const RenderBox& gri
     return masonryGridAreaFromGridAxisSpan(itemSpan);
 }
 
-void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition(const GridTrackSizingAlgorithm& algorithm)
+void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition(const GridTrackSizingAlgorithm& algorithm, std::optional<GridMasonryLayout::OverrideMasonryLogicalWidth> overrideLogicalWidthType)
 {
     for (auto* item : m_itemsWithIndefiniteGridAxisPosition) {
         ASSERT(item);
         if (!item)
             continue;
-        insertIntoGridAndLayoutItem(algorithm, *item, gridAreaForIndefiniteGridAxisItem(*item));
+        insertIntoGridAndLayoutItem(algorithm, *item, gridAreaForIndefiniteGridAxisItem(*item), overrideLogicalWidthType);
     }
 }
 
+LayoutUnit GridMasonryLayout::calculateMasonryIntrinsicLogicalWidth(RenderBox& gridItem, GridMasonryLayout::OverrideMasonryLogicalWidth overrideLogicalWidthType)
+{
+    switch (overrideLogicalWidthType) {
+    case OverrideMasonryLogicalWidth::MinContent:
+        if ((gridItem.style().logicalWidth().isAuto() || gridItem.style().logicalWidth().isPercent())) {
+            if (auto* renderGrid = dynamicDowncast<RenderGrid>(gridItem); renderGrid && renderGrid->isSubgridRows())
+                break;
```

Comment:
Since this logic is repeated in both cases can we just pull it out of the switch statement and put it at the beginning?